### PR TITLE
{Network} `az network vnet`: Migrate to atomic commands

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

Several commands left:
- az network vnet create/update/list-available-ips
- az network peering create/update/sync
- az network subnet create/update/list-available-ips

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -7173,27 +7173,50 @@ def update_vnet_peering(cmd, resource_group_name, virtual_network_name, virtual_
...
-    if subnet.address_prefix is not None:
-        start_ip = subnet.address_prefix.split('/')[0]
-    available_ips = client.virtual_networks.check_ip_address_availability(resource_group_name=resource_group_name,
-                                                                          virtual_network_name=virtual_network_name,
-                                                                          ip_address=start_ip)
-    return available_ips.available_ip_addresses
+    from .aaz.latest.network.vnet import CheckIpAddress
+    from .aaz.latest.network.vnet.subnet import Show
+
+    Show_Subnet = Show(cmd.loader)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7201]:
```suggestion
    Show_Subnet = Show(cli_ctx=cmd.cli_ctx)
```

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -7173,27 +7173,50 @@ def update_vnet_peering(cmd, resource_group_name, virtual_network_name, virtual_
...
+    args = {
+        "resource_group": resource_group_name,
+        "name": subnet_name,
+        "vnet_name": virtual_network_name,
+    }
+    subnet = Show_Subnet(args)
+    if subnet["addressPrefix"]:
+        start_ip = subnet["addressPrefix"].split("/")[0]
+
+    Check_IP_Address = CheckIpAddress(cmd.loader)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7211]:
```suggestion
    Check_IP_Address = CheckIpAddress(cli_ctx=cmd.cli_ctx)
```

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -7173,27 +7173,50 @@ def update_vnet_peering(cmd, resource_group_name, virtual_network_name, virtual_
...
-                                virtual_network_name=virtual_network_name,
-                                subnet_name=subnet_name)
-    if subnet.address_prefix is not None:
-        start_ip = subnet.address_prefix.split('/')[0]
-    available_ips = client.virtual_networks.check_ip_address_availability(resource_group_name=resource_group_name,
-                                                                          virtual_network_name=virtual_network_name,
-                                                                          ip_address=start_ip)
-    return available_ips.available_ip_addresses
+    from .aaz.latest.network.vnet import CheckIpAddress
+    from .aaz.latest.network.vnet.subnet import Show
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7199]: Atomic Commands are all from latest profile, does this function needs to cross profiles?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 4]
```diff
@@ -7173,27 +7173,50 @@ def update_vnet_peering(cmd, resource_group_name, virtual_network_name, virtual_
...
 def list_available_ips(cmd, resource_group_name, virtual_network_name):
-    client = network_client_factory(cmd.cli_ctx).virtual_networks
-    vnet = client.get(resource_group_name=resource_group_name,
-                      virtual_network_name=virtual_network_name)
-    start_ip = vnet.address_space.address_prefixes[0].split('/')[0]
-    available_ips = client.check_ip_address_availability(resource_group_name=resource_group_name,
-                                                         virtual_network_name=virtual_network_name,
-                                                         ip_address=start_ip)
-    return available_ips.available_ip_addresses
+    from .aaz.latest.network.vnet import Show, CheckIpAddress
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:7176]:
```suggestion
    from .aaz.latest.network.vnet import Show, CheckIpAddress
```
Same concern

**Review by kairu-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**